include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -310,6 +423,24 @@ __pattern_adjacent_find(__hetero_tag<_BackendTag>, _ExecutionPolicy&& __exec, _R
     return return_value(result, __rng.size(), __is__or_semantic);
 }
 
+#if _ONEDPL_CPP20_RANGES_PRESENT
+template <typename _BackendTag, typename _ExecutionPolicy, typename _R, typename _Proj, typename _Pred>
+decltype(auto)
```

Comment:
```suggestion
auto
```
It doesn't seem that we need `decltype(auto)` as a return type. Please fix in all applicable places.

Reply:
done